photosynthesis_metrics/utils.py

```diff
@@ -27,18 +27,21 @@ def _validate_input(
         scale_weights: Union[Optional[Tuple[float]], Optional[List[float]], Optional[torch.Tensor]] = None) -> None:
     assert isinstance(x, torch.Tensor) and isinstance(y, torch.Tensor), \
         f'Both images must be torch.Tensors, got {type(x)} and {type(y)}.'
-    assert len(x.shape) == 4, f'Input images must be 4D tensors, got images of shape {x.shape}.'
-    assert x.shape == y.shape, f'Input images must have the same dimensions, got {x.shape} and {y.shape}.'
+    assert 1 < x.dim() < 6, f'Input images must be 2D, 3D or 4D tensors, got images of shape {x.size()}.'
+    assert x.size() == y.size(), f'Input images must have the same dimensions, got {x.size()} and {y.size()}.'
+    if x.dim() == 5:
+        assert x.size(-1) == 2, f'Expected Complex 5D tensor with (N,C,H,W,2) size, got {x.size()}'
     if kernel_size is not None:
         assert kernel_size % 2 == 1, f'Kernel size must be odd, got {kernel_size}.'
     if scale_weights is not None:
         assert isinstance(scale_weights, (list, tuple, torch.Tensor)), \
             f'Scale weights must be of type list, tuple or torch.Tensor, got {type(scale_weights)}.'
-        assert len(scale_weights) == 5, f'Scale weights collection must contain 5 values, got {len(scale_weights)}.'
+        assert (torch.Tensor(scale_weights).dim() == 1), \
```

Comment:
It's better to use `torch.tensor` function instead of `torch.Tensor` for converting into tensors, because output dtype is same as inputs. In this particular case result will be the same `torch.float32`, but sometimes it can lead to types mismatch.
See https://discuss.pytorch.org/t/difference-between-torch-tensor-and-torch-tensor/30786 for details